File: leetcode/2610. Convert an Array Into a 2D Array With Conditions.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        vector<vector<int>> ans;
        vector<int> count(nums.size() + 1);

        for(int num : nums){
            if(ans.size() < ++count[num]){
                ans.push_back({});
            }
            ans[count[num] - 1].push_back(num);
        }
        return ans;
    }
};

File: leetcode/2610. Convert an Array Into a 2D Array With Conditions.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        ans = []
        freq_count = [0 for i in range(len(nums) + 1)]

        for num in nums:
            if len(ans) < freq_count[num] + 1:
                ans.append([])
            
            ans[freq_count[num]].append(num)
            freq_count[num] += 1
        
        return ans